src/app/admin/orders/page.tsx

```
@@ -1,68 +1,244 @@
+        const fullName = customer ? `${customer.first_name || ''} ${customer.last_name || ''}`.trim() : '';
+        return fullName || '\u00A0'; // Render non-breaking space if fullName is empty
+      },
+    },
+    {
+      id: 'shipping_address_formatted', // Keep unique id
+      accessorKey: 'shipping_address_id', // Access the shipping_address_id
+      header: 'عنوان الشحن', // Keep header
+      cell: ({ row }: { row: { original: Order } }) => {
+        const addressId = row.original.shipping_address_id; // Get the address ID
+        const [address, setAddress] = useState<AddressWithPhone | null>(null);
+        const [loadingAddress, setLoadingAddress] = useState(true);
```

Biome (2.1.2)

[error] This hook is being called from a nested function, but all hooks must be called unconditionally from the top-level component.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)